### PR TITLE
Replace bash script lftp upload with ftplib upload

### PR DESCRIPTION
**Pull request by ghost**
- Remove lftp file upload bash script from directory
- Remove call to lftp upload bash script from python script
- Use ftplib to establish FTPS with shared TLS session 
- Upload .bw files to usegalaxy.org over connection

### CODE REVIEW

File: upload.py [hunk 1]
```diff
@@ -17,11 +17,32 @@
 import glob
 import argparse
 import getpass
+from ftplib import FTP_TLS
+import os
+
+import ssl
+import ftplib
+from ftplib import FTP
+from ftplib import FTP_TLS
```
Comment (fabubaker) [inline upload.py]: Let's try to remove redundant imports.

File: upload.py [hunk 2]
```diff
@@ -17,11 +17,32 @@
...
 
 import subprocess as sp
```
Comment (fabubaker) [inline upload.py:18]: Don't think we need `subprocess` anymore.

File: upload.py [hunk 3]
```diff
@@ -83,7 +115,7 @@
     
 track_file.close()
 
-print(">>> Uploading track file to FTP and generating link...")
+print (">>> Uploading track file to FTP and generating link...")
 
 p = sp.call(["lftp", "-c", "open -u %s,%s usegalaxy.org; put %s" % (args.username, password, track_filename)])
```
Comment (fabubaker) [inline upload.py]: Shouldn't this `lftp` call be replaced with the correct `ftplib` call?

**Review by fabubaker — CHANGES_REQUESTED** [review]

File: upload.py [hunk 4]
```diff
@@ -72,15 +72,16 @@ def ntransfercmd(self, cmd, rest=None):
...
+ftps = FTPS_connect (host='usegalaxy.org', user=args.username, passwd=password)
 ftps.set_debuglevel(10)
 ftps.prot_p()
 ftps.retrlines('LIST')
 
 # Upload all .bw filenames to the galaxy FTP server.
 for filename in (glob.glob(args.dir + "/*.bw")):
     f = open(filename, 'rb')
-    cmd = 'STOR '+filename
+    bwname = filename.split("/")[-1]
```
Comment (fabubaker) [inline upload.py]: `os.path.basename(path)` is better than manually splitting and extracting the filename.

**Review by fabubaker — CHANGES_REQUESTED** [review]

**Review by fabubaker — CHANGES_REQUESTED** [review]
Can you remove `lftp` from README.md and the Dockerfile?

File: upload.py [hunk 5]
```diff
@@ -44,19 +55,29 @@
...
 
 # First upload all .bw files to the galaxy FTP server.
-# NOTE: ftplib has issues with galaxy FTP, hence the lftp script.
 print(">>> Uploading .bw files to FTP server.")
-print(">>> Executing ./lftp_upload.sh %s %s %s" % (args.username, "*****", args.dir))
-p = sp.call(["./lftp_upload.sh", args.username, password, args.dir])
+
+# Set up FTPS connection
+ftps = FTPS_connect (host="usegalaxy.org", user=args.username, passwd=password)
+ftps.set_debuglevel(10)
```
Comment (fabubaker) [inline upload.py]: This prints a lot of verbose debug output. Let's remove this line.

File: upload.py [hunk 6]
```diff
@@ -44,19 +55,29 @@
...
+ftps.prot_p()
+ftps.retrlines("LIST")
```
Comment (fabubaker) [inline upload.py]: This just lists the contents of the directory on the FTP server. We can remove it too.

**Review by fabubaker — CHANGES_REQUESTED** [review]

File: upload.py [hunk 7]
```diff
@@ -44,19 +55,29 @@
...
+
+# Upload all .bw filenames to the galaxy FTP server.
+for filename in (glob.glob(args.dir + "/*.bw")):
+    f = open(filename, "rb")
+    bwname = os.path.basename(filename)
+    cmd = "STOR " + bwname
+    ftps.storbinary (cmd, f)
```
Comment (fabubaker) [inline upload.py:74]: Add a print statement here saying something like `Uploading {filename}.` Helps keep track of progress when you're uploading a lot of files.

**Review by fabubaker — CHANGES_REQUESTED** [review]

File: upload.py (hunk 5 above)
Comment (ghost) [reply upload.py]: Are you talking about line 66 only?

Comment (fabubaker) [reply upload.py]: Yes.

**Review by fabubaker — APPROVED** [review]

**Pull request merged**